### PR TITLE
Release 4.9.0

### PR DESCRIPTION
**Pull request by gongstr**
# Release 4.9.0
Added properties on event
Additional bug fixes and improvements

## Are Network Changes Included?

Does this change introduce new network connections to any UserLeap resources or third parties?

- [ ] Yes
- [ ] No

If yes, have you taken every precaution to limit potential data exposure using best-practice encryption and secure connection protocols?

- [ ] Yes
- [ ] No

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by huangjoyce3 — APPROVED** [review]

File: UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js [hunk 1]
```diff
@@ -20,12 +20,17 @@ function configure(environmentId, mobileHeadersJSON) {
 }
 
 function handleSurveyCallback(surveyState, callbackId) {
+    // no survey handler but the survey is shown
+    if (!callbackId) {
+        Sprig.dismissActiveSurvey()
```
Comment (helloandrewpark) [inline UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js:25]: nit: missing ;

File: UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js [hunk 2]
```diff
@@ -42,10 +47,11 @@ Sprig('addListener', 'sdk.ready', () => {
 Sprig('addListener', 'survey.height', (payload) => {
     window.webkit.messageHandlers.sprigWebController.postMessage({type: 'setHeight', height: payload.contentFrameHeight.toString() });
 });
-Sprig.mobileTrackEvent = async (event, userId, partnerAnonymousId, callbackId) => {
+Sprig.mobileTrackEvent = async (event, userId, partnerAnonymousId, properties, callbackId) => {
     const payload = { eventName: event };
     if (userId) payload.userId = userId;
     if (partnerAnonymousId) payload.anonymousId = partnerAnonymousId;
+    if (properties) payload.properties = properties;
```
Comment (helloandrewpark) [inline UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js:54]: should we also check that it's an object with key count > 0?

**Review by helloandrewpark — APPROVED** [review]
added some nits

File: UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js (hunk 1 above)
Comment (gongstr) [reply UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js:25]: will roll the nits into the next release

File: UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js (hunk 2 above)
Comment (huangjoyce3) [reply UserLeapKit.xcframework/ios-arm64_x86_64-simulator/UserLeapKit.framework/sprig_web_assets.bundle/snippet.js:54]: the API will check if it is empty before processing

**Pull request merged**